Sources/BetterSafariView/SafariView/SafariViewPresenter.swift

```diff
@@ -79,20 +79,31 @@ extension SafariViewPresenter {
             let safariViewController = SFSafariViewController(url: representation.url, configuration: representation.configuration)
             safariViewController.delegate = self
             representation.applyModification(to: safariViewController)
-            
-            // Present a Safari view controller from the `viewController` of `UIViewRepresentable`, instead of `UIViewControllerRepresentable`.
-            // This fixes an issue where the Safari view controller is not presented properly
-            // when the `UIViewControllerRepresentable` is detached from the root view controller (e.g. `UIViewController` contained in `UITableViewCell`)
-            // while allowing it to be presented even on the modal sheets.
-            // Thanks to: Bohdan Hernandez Navia (@boherna)
-            guard let presentingViewController = uiView.viewController else {
-                self.resetItemBinding()
+
+            if #available(iOS 17.0, *) {
+                if var topController = UIApplication.shared.windows.filter(\.isKeyWindow).first?.rootViewController {
```

Comment:
An `UIWindow` instance is must be grabbed via `uiView.window`.

`UIApplication.shared.windows` is deprecated in iOS 15, and this will make an issue when there are two or more `UIWindowScene`s.